### ISSUE TITLE
gloo error: ‘memcpy’ was not declared in this scope

### ISSUE THREAD

**Issue by akakcolin** [facebookarchive/caffe2#286]
```
caffe2/third_party/gloo/gloo/mpi/context.cc: In member function ‘void gloo::mpi::Context::connectFullMesh(std::shared_ptr<gloo::transport::Device>&)’:
caffe2/third_party/gloo/gloo/mpi/context.cc:82:71: error: ‘memcpy’ was not declared in this scope
     memcpy(in.data() + (i * maxLength), address.data(), address.size());
                                                                       ^
caffe2/third_party/gloo/gloo/mpi/context.cc:100:58: error: ‘memcpy’ was not declared in this scope
     memcpy(address.data(), out.data() + offset, maxLength);

```
fix the problem by including the header 
`#include <cstring>`

**Comment by Yangqing** [comment]
Thanks! cc @pietern who is the lead developer on Gloo.

**Comment by pietern** [comment]
Thanks for the report, fixing.

**Comment by pietern** [comment]
In the mean time you can "fix" by setting USE_MPI to OFF. This only happens on the intersection of USE_GLOO and USE_MPI.

**Comment by Yangqing** [comment]
Fixed in 99db0cb5f68a623642782cfc8088e28fe49a1d59

Thanks!
